### ISSUE TITLE
Only able to use GPU 0 for chainer computation

### ISSUE THREAD

**Issue by sidak** [pfnet/chainer#1349]
I am using centos 6.5 and I have installed chainer. I have also installed cudnn v5 (placed .h and .so files in required places) and also have cuda 7.5 and Nvidia driver 352  installed. 

When I try to run chainer code like: 

with cupy.cuda.Device(GPU_ID): 

 .....................................

It works only when GPU_ID = 0. But doesn't work for other GPU's. (I have a total of 10 gpu's on my machine)

I get the following error:

Traceback (most recent call last):
  File "cifar10_dcgan.py", line 495, in <module>
    opt.update(model, z, x)
  File "/home/sidak/Documents/installation_python/lib/python2.7/site-packages/chainer/optimizer.py", line 376, in update
    loss = lossfun(_args, *_kwds)
  File "cifar10_dcgan.py", line 210, in **call**
    self.X = self.sample(z) # size: BATCH_SIZE x 3072
  File "cifar10_dcgan.py", line 183, in sample
    h = F.reshape(F.relu(self.bn0l(self.l0z(z))), (z.data.shape[0], 512, 4, 4 ))
  File "/home/sidak/Documents/installation_python/lib/python2.7/site-packages/chainer/links/connection/linear.py", line 72, in **call**
    return linear.linear(x, self.W, self.b)
  File "/home/sidak/Documents/installation_python/lib/python2.7/site-packages/chainer/functions/connection/linear.py", line 79, in linear
    return LinearFunction()(x, W, b)
  File "/home/sidak/Documents/installation_python/lib/python2.7/site-packages/chainer/function.py", line 130, in **call**
    outputs = self.forward(in_data)
  File "/home/sidak/Documents/installation_python/lib/python2.7/site-packages/chainer/functions/connection/linear.py", line 39, in forward
    y += b
  **File "cupy/core/core.pyx", line 898, in cupy.core.core.ndarray.**iadd** (cupy/core/core.cpp:18222)
  File "cupy/core/elementwise.pxi", line 728, in cupy.core.core.ufunc.**call** (cupy/core/core.cpp:38845)
  File "cupy/core/elementwise.pxi", line 99, in cupy.core.core._preprocess_args (cupy/core/core.cpp:27436)
ValueError: Array device must be same as the current device: array device = 0 while current = 7**

I have tried a lot but I am not able to solve this problem. Thank you so much for your help!

**Comment by chenzhekl** [comment]
I guess it may have something to do with the device your `Variable` and model are allocated on. If they are not allocated on the same place, such an error may occur.

**Comment by sidak** [comment]
Yes, I tried it a few days ago and it worked after that. But in my past experiments, if I wrote `with cupy.cuda.DEVICE(GPU_ID)` before using variables, it used to work. Now, I had to explicitly set them to GPU using the method to_gpu. 
I don't know why it should behave this way, but it might be due to  the updates that I had installed for cuda  and the driver. 

Anyways, thank you for your help 😄  ! @chenzhekl
